source/gear/swordofthesun-lethal.js

```diff
@@ -0,0 +1,44 @@
+const { GearTemplate } = require('../classes');
+const { ESSENCE_MATCH_STAGGER_FOE } = require('../constants');
+const { getModifierCategory } = require('../modifiers/_modifierDictionary');
+const { dealDamage, changeStagger, combineModifierReceipts, removeModifier, generateModifierResultLines } = require('../util/combatantUtil');
+const { damageScalingGenerator } = require('./shared/scalings');
+
+module.exports = new GearTemplate("Lethal Sword of the Sun",
+	[
+		["use", "Removes all buffs from a foe, then deal <@{damage}> @{essence} damage to a them, plus 30 damage per unique buff removed"],
+		["critical", "Damage x @{critBonus}"]
+	],
+	"Offense",
+	"Fire"
+).setCost(450)
+	.setEffect((targets, user, adventure) => {
+		totalResultLines = [];
+		for (const target of targets) {
+			const targetBuffs = Object.keys(target.modifiers).filter(modifier => getModifierCategory(modifier) === "Buff");
+			removedBuffReceipts = []
+			targetBuffs.forEach(buffName => {
+				removedBuffReceipts.push(...removeModifier([target], { name: buffName, stacks: "all" }))
+			})
+			totalResultLines.push(...generateModifierResultLines(combineModifierReceipts(removedBuffReceipts)));
```

Comment:
Moving this push to after the damage result lines would better comply with the result line convention of damage lines showing up first (also applies to all three variants).

Reply:
hrm.... the removal mechanically preceding damage is pretty important though. It's what lets you bunker-bust through defensive buffs.

Reply:
The removal still happens first (during the addModifier), push is a text side-effect.